### PR TITLE
FEATURE: make media browser uri path configurable

### PR DESCRIPTION
**Pull request by so-grimm**
- add and use new data-attribute and const for mediaBrowserUriPath with fallback value of '/neos/media/browser'
- add explanation for new data-attribute to Readme

Todo:
- [x]  Minify js

### CODE REVIEW

File: Resources/Private/source.js [hunk 1]
```diff
@@ -135,7 +135,8 @@ window.addEventListener('DOMContentLoaded', () => {
 
 		const multiple = container.dataset.assetMultiple !== undefined;
 		const assetIdentifier = container.dataset.asset;
-		const url = assetIdentifier && !multiple ? new URL('/neos/media/browser/images/edit.html?asset=' + assetIdentifier, window.location.origin) : new URL('/neos/media/browser/assets/index.html', window.location.origin);
+		const mediaBrowserUriPath = container.dataset.assetMediaBrowserUriPath || '/neos/media/browser';
+		const url = assetIdentifier && !multiple ? new URL(mediaBrowserUriPath + '/images/edit.html?asset=' + assetIdentifier, window.location.origin) : new URL(mediaBrowserUriPath + '/assets/index.html', window.location.origin);
```
Comment (ahaeslich) [inline Resources/Private/source.js]: Looking at the code of this plugin: If we respect the (potential) subfolder here. Isn't it missing in line 39?

Comment (bwaidelich) [reply Resources/Private/source.js]: The comment is now marked outdated but there are only two commits from 1 h ago!? @so-grimm did you force-push to the branch?

**Comment by bwaidelich** [comment]
@so-grimm Thanks for this – I tried to push a tweak to the README and the minified JS but I'm not allowed to. Could you "Allow edits by maintainers"?

File: README.md [hunk 2]
```diff
@@ -36,6 +36,7 @@ Neos Media Browser and/or asset preview:
 | `asset-constraints-media-types`   | Comma separated list of allowed mediaTypes (optional)                                                   | container                                                         |
 | `asset-constraints-asset-sources` | Comma separated list of allowed asset sources (optional)                                                | container                                                         |
 | `asset-multiple`                  | If set, multiple assets can be added (optional)                                                         | container                                                         |
+| `asset-neos-uri-path`             | Will change the uri path. I.e. necessary when working with subfolders                                   | container                                                         |
```
Comment (bwaidelich) [inline README.md]:
```suggestion
| `asset-neos-uri-path`             | Can be set if the Neos routes don't reside underneath `/neos` (optional)                                | container                                                         |
```

**Review by bwaidelich — COMMENTED** [review]
I added my suggestions as inline suggestion now.

For the `dist.js` you can just use https://www.toptal.com/developers/javascript-minifier and past the source code of `dialog-polyfill.js` and `source.js`

File: Resources/Private/source.js (hunk 1 above)
Comment (ahaeslich) [reply Resources/Private/source.js]: This comment was made on the first commit. The second commit did change these lines ... Shouldn't they be outdated now?

**Comment by so-grimm** [comment]
@bwaidelich I just found out that it's actually not possible to "Allow edits by maintainers" if the PR is coming from an organisation and not an individual. But Anke gave you write-access to the repo, so you should be able to fix whatever you need to now.

File: Resources/Private/source.js (hunk 1 above)
Comment (bwaidelich) [reply Resources/Private/source.js]: yes, you're right of course. I misinterpreted the commit log

**Comment by bwaidelich** [comment]
@so-grimm @ahaeslich Thanks a lot!
=> https://github.com/bwaidelich/Wwwision.Neos.MediaBrowserDialog/releases/tag/2.4.0

**Pull request merged**